### ISSUE TITLE
[Question] apply EventTermCfg vs XML modification

### ISSUE THREAD

**Issue by Jaebeom-git** [mujocolab/mjlab#488]
I compared two approaches to apply a +2 kg mass increase to the G1 torso with proportional inertia scaling:
1. Static XML modification
```xml
<inertial pos="0.00203158 0.000339683 0.184568" quat="0.999803 -6.03319e-05 0.0198256 0.00131986" mass="9.818" diaginertia="0.153039832 0.1379402 0.034381116"/>
<!-- <inertial pos="0.00203158 0.000339683 0.184568" quat="0.999803 -6.03319e-05 0.0198256 0.00131986"
  mass="7.818" diaginertia="0.121847 0.109825 0.0273735"/> -->
```
3. Runtime configuration using a custom randomize_body_mass_with_inertia event, applied at startup as follows:
```python
"add_mass": EventTermCfg(
    func=mdp.randomize_body_mass_with_inertia,
    mode="startup",
    domain_randomization=True,
    params={
        "asset_cfg": SceneEntityCfg("robot", body_names=("torso_link")),
        "operation": "add",
        "field": "body_mass",
        "distribution": "uniform",
        "mass_range": (2.0, 2.0),
    },
),
```

I ran training for 10 steps and observed a clear difference in training behavior:

1. Editing XML: Mean Reward ≈ -3.0, Angular Momentum Penalty ≈ -0.03.
2. Runtime Configuration: Mean Reward ≈ -80.0, Angular Momentum Penalty ≈ -4.0.

I would like to ask for advice on the possible cause of this discrepancy.

The runtime mass and inertia modification is implemented as follows:

```python
def randomize_body_mass_with_inertia(
    env: ManagerBasedRlEnv,
    env_ids: torch.Tensor | None,
    mass_range: tuple[float, float],
    operation: Literal["add", "scale", "abs"] = "add",
    distribution: Literal["uniform", "log_uniform", "gaussian"] = "uniform",
    asset_cfg: SceneEntityCfg = _DEFAULT_ASSET_CFG,
    min_mass: float = 1e-6,
    **kwargs,
):
    if env_ids is None:
        env_ids = torch.arange(env.num_envs, device=env.device, dtype=torch.int)
    else:
        env_ids = env_ids.to(env.device, dtype=torch.int)

    asset = env.scene[asset_cfg.name]
    asset_cfg.resolve(env.scene)

    body_ids = asset.indexing.body_ids[asset_cfg.body_ids]

    default_mass_all = env.sim.get_default_field("body_mass")
    default_inertia_all = env.sim.get_default_field("body_inertia")

    default_mass = default_mass_all[body_ids]
    default_inertia = default_inertia_all[body_ids]

    default_mass = default_mass.unsqueeze(0).expand(len(env_ids), -1)

    shape = default_mass.shape
    lower, upper = mass_range

    if distribution == "uniform":
        noise = sample_uniform(lower, upper, shape, env.device)
    elif distribution == "log_uniform":
        noise = sample_log_uniform(lower, upper, shape, env.device)
    elif distribution == "gaussian":
        noise = sample_gaussian(lower, upper, shape, env.device)
    else:
        raise ValueError(distribution)

    if operation == "add":
        new_mass = default_mass + noise
    elif operation == "scale":
        new_mass = default_mass * noise
    elif operation == "abs":
        new_mass = noise
    else:
        raise ValueError(operation)

    new_mass = torch.clamp(new_mass, min=min_mass)

    ratio = new_mass / (default_mass + 1e-9)
    new_inertia = default_inertia.unsqueeze(0) * ratio.unsqueeze(-1)

    env_grid, body_grid = torch.meshgrid(env_ids, body_ids, indexing="ij")
    env.sim.model.body_mass[env_grid, body_grid] = new_mass
    env.sim.model.body_inertia[env_grid, body_grid] = new_inertia
```

**Comment by kevinzakka** [comment]
Hi @Jaebeom-git,

Mass is considered a "complex" randomization because changing it requires updating dependent model fields. `geom_friction` is "simple": you can modify it directly without any dependencies.

For mass specifically, when you change `body_mass`, you also need to:
1. Update `body_inertia` (which your function already does correctly)
2. Update `body_subtreemass` (sum of masses in the kinematic subtree)

Your custom `randomize_body_mass_with_inertia` function handles 1. but not 2. which makes the physics inconsistent.

Once https://github.com/google-deepmind/mujoco_warp/pull/905 lands in mujoco_warp, we'll be able to implement this properly by calling the equivalent of `mj_setConst` to recompute `body_subtreemass`.

BTW, you might want to read [this part of the MuJoCo documentation](https://mujoco.readthedocs.io/en/latest/programming/simulation.html#mjmodel-changes) for some more context.

**Comment by Jaebeom-git** [comment]
Hi @kevinzakka,

Thanks for the clarification and for pointing me to the relevant MuJoCo documentation!! :D 👍
